### PR TITLE
Create Enumerable Methods

### PR DESCRIPTION
**Pull request by gisachris**
## IN This PR We present The  Enumerable Creation :
 
Below are some of my implementations as well as going over the project requirements 🤝 .
## Requirements
- Setup ruby workflow and linters 🔍 
- use GitFlow 🌊.
- Create a class `MyList` that has an instance variable `@list`.
- In `MyList` implement a method `#each` that yields successive members of `@list` and uses the `MyEnumerable` module.
- Create a module `MyEnumerable` that implements the following methods :
    - `#all?`
    - `#any?`
    - `#filter`

## Achievements
- Setup ruby workflow and linters 🔍✅ 
- use GitFlow 🌊.✅
- Create a class `MyList` that has an instance variable `@list`.✅
- In `MyList` implement a method `#each` that yields successive members of `@list` and uses the `MyEnumerable` module.✅
- Create a module `MyEnumerable` that implements the following methods✅

## Conclusion
**We are looking forward to hearing 👂  from you Our CR in terms of what can be better implemented 👨‍💻  for the project thankyou!. And we look forward to hearing from you soon.**  🙇

### CODE REVIEW

File: list.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require_relative 'enumerable'
+
+class MyList
+  include MyEnumerable
+
+  def initialize(*elems)
+    @list = elems
```
Comment (t-yanick) [inline list.rb:7]: - [ ] Great job working on this milestone. When I run ruby main.rb from the terminal, I get error and when I run ruby list.rb i get no response. I am unable to start your application. Kindly review and if possible provide instructions in your README how a user can start the application.
Screenshot from 2023-07-28 09-05-42

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @gisachris and @toyman640 , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Use of `require_relative` to import `MyEnumerable`
:heavy_check_mark: `MyEnumerable` module  implements `#all?` `#any?` and `#filter` method

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: list.rb (hunk 1 above)
Comment (gisachris) [reply list.rb:7]: @t-yanick Thank you for the great insight and review! we will solve the bug and send it for another review.

**Comment by gisachris** [comment]
## IN This PR We present Changes♻️ to Enumerable Creation :
 
Below are some of my implementations as well as going over the project requirements 🤝 .
## Changes Required
- Setup clear documentation on how to use the app 👨‍💻 as per reviewer request [here](https://github.com/gisachris/Enumerables/pull/1#discussion_r1277252937) 
- Fix other bugs 🐞 

## Achievements
- Setup clear documentation✅ 
- Fix bugs.✅

## Conclusion
**We are looking forward to hearing 👂  from you Our CR in terms of what can be better implemented 👨‍💻  for the project thankyou!. And we look forward to hearing from you soon.**  🙇

**Review by ghost — APPROVED** [review]
Hi @gisachris , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ Good work fixing the bug👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**